pkg/providers/amifamily/windows.go

```diff
@@ -0,0 +1,89 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package amifamily
+
+import (
+	"fmt"
+
+	"github.com/aws/karpenter-core/pkg/scheduling"
+
+	"github.com/samber/lo"
+	"k8s.io/apimachinery/pkg/api/resource"
+
+	"github.com/aws/aws-sdk-go/aws"
+
+	"github.com/aws/karpenter/pkg/providers/amifamily/bootstrap"
+
+	v1 "k8s.io/api/core/v1"
+
+	"github.com/aws/karpenter-core/pkg/apis/v1alpha5"
+	"github.com/aws/karpenter-core/pkg/cloudprovider"
+
+	"github.com/aws/karpenter/pkg/apis/v1alpha1"
+)
+
+type Windows struct {
+	DefaultFamily
+	*Options
+}
+
+func (w Windows) DefaultAMIs(version string) (results []DefaultAMIOutput) {
+	for _, windowsVersion := range v1alpha1.SupportedWindowsVersions {
+		for _, windowsVariant := range v1alpha1.SupportedWindowsVariants {
+			results = append(results, DefaultAMIOutput{Name: fmt.Sprintf("windows-server-%s-%s-%s", windowsVersion, windowsVariant, version),
+				Query: fmt.Sprintf("/aws/service/ami-windows-latest/Windows_Server-%s-English-%s-EKS_Optimized-%s/image_id", windowsVersion, windowsVariant, version),
+				Requirements: scheduling.NewRequirements(
+					scheduling.NewRequirement(v1.LabelArchStable, v1.NodeSelectorOpIn, v1alpha5.ArchitectureAmd64),
+					scheduling.NewRequirement(v1.LabelOSStable, v1.NodeSelectorOpIn, string(v1.Windows)),
+					scheduling.NewRequirement(v1alpha1.LabelWindowsVersion, v1.NodeSelectorOpIn, windowsVersion),
```

Comment:
Can these scheduling requirements be based on the [upstream scheduling requirements](https://kubernetes.io/docs/concepts/windows/user-guide/#ensuring-os-specific-workloads-land-on-the-appropriate-container-host) that users are probably already putting on their workloads?

Reply:
Reasonable
But ssm has no such mapping

Reply:
The current upstream well-known labels don't meet the Windows AMI selecting requirements.
For pods, it may be enough (however, from my experience, a Kubernetes admin still needs to know the detail of the workload).
For EKS AMI, it needs three arguments to query from SSM
https://docs.aws.amazon.com/eks/latest/userguide/retrieve-windows-ami-id.html
Windows_Server-${2019/2022}-English-${Core/Full}-EKS_Optimized-${k8s_version}